core/client/db/src/lib.rs

```diff
@@ -896,11 +896,18 @@ impl<Block> client::backend::Backend<Block, Blake2Hasher> for Backend<Block> whe
 		Some(&self.changes_tries_storage)
 	}
 
-	fn revert(&self, n: NumberFor<Block>) -> Result<NumberFor<Block>, client::error::Error> {
+	fn revert(&self, m: NumberFor<Block>) -> Result<NumberFor<Block>, client::error::Error> {
 		use client::blockchain::HeaderBackend;
+
 		let mut best = self.blockchain.info()?.best_number;
-		// if the best is lower to n(less then 256), just use best number in case overflow
-		let n = if best < n { best } else { n };
+		let finalized = self.blockchain.info()?.finalized_number;
+		let reversible = best - finalized;
+		let mut n = m;
+
+		if reversible < m { 
+			n = reversible;
+			info!("There are only {} non finalized blocks to revert.", n);
```

Comment:
I would like to have this output at the point, where we print how many blocks are reverted. In the case that we do not revert any blocks, because there are no non finalized. I would like to have one message that states this and the `0 blocks` reverted would not be printed at all.

Reply:
Fixed.